src/CircularSlider.js

```diff
@@ -129,26 +156,59 @@ export default class CircularSlider extends PureComponent {
 
       onPanResponderMove: (evt, { moveX, moveY }) => {
         const { circleCenterX, circleCenterY } = this.state;
-        const { startAngle, onUpdate } = this.props;
+        const { startAngle, onUpdate, circlePercentInDecimal } = this.props;
+
+        let newAngle =
+          Math.atan2(moveY - circleCenterY, moveX - circleCenterX) +
+          Math.PI / 2;
 
-        let newAngle = Math.atan2(moveY - circleCenterY, moveX - circleCenterX) + Math.PI / 2;
         let newAngleLength = (newAngle - startAngle) % (2 * Math.PI);
 
         if (newAngleLength < 0) {
           newAngleLength += 2 * Math.PI;
         }
-        onUpdate({ startAngle, angleLength: newAngleLength });
+
+        let aNewAngle =
+          Math.atan2(moveY - circleCenterY, moveX - circleCenterX) +
+          Math.PI / 2 +
+          (1 - circlePercentInDecimal) *
+            2 *
+            Math.PI *
+            (newAngleLength / (2 * Math.PI));
+        let aNewAngleLength = (aNewAngle - startAngle) % (2 * Math.PI);
+
+        if (aNewAngleLength < 0) {
```

Comment:
Is there a way you can name newAngle and aNewAngle so it's easier to understand what they are and how they are different?  Or perhaps add a comment to explain?  Might make it easier if anyone else ever wants to come in and modify.

Reply:
`Might make it easier if anyone else ever wants to come in and modify.` Look at you, thinking about future developers.